admin/src/main/resources/io/buoyant/admin/js/src/router_services.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+define([
+  'jQuery',
+  'template/compiled_templates',
+  'src/router_service',
+  'bootstrap'
+], function($,
+  templates,
+  RouterService
+) {
+  var svcsByRouter = {};
+
+  function extractSvcClientStats(data, router) {
+    return _.chain(_.get(data, ["rt", router, "client"]))
+      .map(function(clientData, client) {
+        return _.map(clientData.service, function(svcData, svc) {
+          return { service: svc, client: client, requests: svcData.requests.delta };
+        });
+      })
+      .flatten()
+      .groupBy("service")
+      .value();
+  }
+
+  function initializeRouterServices(metricsCollector, initialData, $container) {
+    _.each(initialData, function(routerData, router) {
+      svcsByRouter[router] = svcsByRouter[router] || {};
+
+      var $routerContainer = $(templates.router_services_container({
+        router: router
+      }));
+      $container.append($routerContainer);
+
+      if(_.isEmpty(routerData.services)) {
+        $routerContainer.find(".svc-subsection").addClass("hidden");
+      }
+
+      _.each(routerData.services, function(svc) {
+        initializeService($routerContainer, metricsCollector, router, svc);
+      });
+    });
+  }
+
+  function initializeService($routerContainer, metricsCollector, router, svc) {
+    $routerContainer.find(".svc-subsection").removeClass("hidden");
```

Comment:
I'd rather not do a DOM find for every added service if we don't need to (and I don't think we do for initializeServices). For addServices, you really only need to check size at the end, maybe something like `_.flatten(svcsByRouter).length > 0`